diesel_derives/Cargo.toml

```diff
@@ -9,11 +9,7 @@ repository = "https://github.com/diesel-rs/diesel/tree/master/diesel_derives"
 autotests = false
 include = ["src/**/*", "LICENSE-*"]
 rust-version = "1.65.0"
-# we cannot update to newer editions
-# because we relay on a name resolution
-# hack to get a `diesel` item inside of diesel
-# as well as for using diesel-derives in external crates
-edition = "2015"
+edition = "2021"
```

Comment:
We need this 2015, right?

Reply:
We needed it before https://github.com/diesel-rs/diesel/pull/3564
I tried that migration already yesterday and noticed that dependencies. I did not think much about that and went on trying to minimize one of the rust-analyzer type issues with diesel, so that the rust-analyzer team had at least a minimal example without depending on diesel. While doing that someone from that team said that they cannot explain how name resolution even worked for code generated by diesel proc-macro. Because I hit that by accident a bit earlier I was able to explain that and they suggested the `extern crate self as diesel;` fix. That in turn remove the need to use the 2015 edition there, so after all that's all somehow related.

Reply:
Weird, I remember trying this exact trick when I was refactoring the derives. But, it didn't work back then, so I thought maybe the issue was something else and ended up keeping the old hack.